packages/@aws-cdk/aws-iam/lib/role.ts

```diff
@@ -137,6 +159,16 @@ export class Role extends Construct implements IIdentityResource, IPrincipal, ID
     this.defaultPolicy.addStatement(statement);
   }
 
+  public findOrAddStatement(statement: PolicyStatement): PolicyStatement {
```

Comment:
This feels like a tricky one. What if the statement I want to add is different than the one already in the policy? It's not clear what this is doing in that case? Some people may expect those to be merged somehow.

I rather we just have `tryFindStatement(sid)` and then users can decide how to deal with the result.

Reply:
I thought about this. And it felt like a race condition waiting to happen. I guess I can do that, though... Found it makes the usage a bit cumbersome (but fine, it's not like you need to do this often anyway).

I thought about providing stronger guarantees (aka the corresponding statement to match all non-`undefined` attributes of the "default" installment), but the `PolicyStatement` class is currently designed to be write-only...

Reply:
Keep it simple. We are single threaded. I don't see any potential race conditions if the user writes:

```
s = tryFind()
if (!s) {
  add(new S())
} else {
  s.update()
}
```

Reply:
> We are single threaded

Right now we are. We don't have to always be. Ruby has been "single threaded" (by virtue of a GIL), wants to get rid of the GIL and is basically unable to because *everything* assumes (either knowingly or unknowingly) there is one. I don't want to be that guy. Aaaaanyway, as I said, I'm not touching this part anymore, so win win.